### ISSUE TITLE
Error No p element in scope but a p end tag seen

### ISSUE THREAD

**Issue by rroobbeerr** [validator/validator#1956]
### Summary

This code fails, it does not recognize the start 'p':

```
<!DOCTYPE html>
<html lang="es">
<head>
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<title>Title</title>
</head>
<body>
<p>
  <ul>
    <li>Text
  </ul>
</p>
</body>
</html>
```

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!DOCTYPE html>
<html lang="es">
<head>
	<meta http-equiv="content-type" content="text/html; charset=utf-8">
	<title>Title</title>
</head>
<body>
<p>
  <ul>
    <li>Text
  </ul>
</p>
</body>
</html>
```

### Checker error messages

```shell
Error: No p element in scope but a p end tag seen.

    From line 12, column 1; to line 12, column 4

    t↩  </ul>↩</p>↩</bod
```

### Steps to reproduce

n/a

### Expected behavior

no errors

### Actual behavior

Error: No p element in scope but a p end tag seen.

    [From line 12, column 1; to line 12, column 4](https://validator.w3.org/nu/#l12c4)

    t↩  </ul>↩</p>↩</bod

**Comment by sideshowbarker** [comment]
Thanks for reporting this but what’s described in the issue description is actually expected behavior.

The reason is: A `ul` element can’t be placed within a `p` element. Or to put it in other terms: If you have the following markup:

```html
<p>
  <ul>
    <li>Text
  </ul>
</p>
```
…then what that actually represents in a (parsed) HTML document is: A `p` element _followed by_ a `ul` element, followed by _another_ `p` element.

`</p>` end tags are allowed to be omitted under certain circumstances. In other terms, under those circumstances, the `</p>` end tag is _implied_ by the markup.  And so another way to state what’s going on  is, the `<p>` start tag in that markup is followed _immediately_ by an _implied_ `</p>` end tag.

And the reason for that is: The HTML parsing rules are such that, when a `p` element is open, and the parser then finds a `<ul>` start tag, the parser closes the `p` element. Or another way to consider it conceptually is that the parser generates an implied `</p>` end tag.

That can all be verified with these steps:

1. Put the markup source from the issue description into a `foo.html` (or whatever name) file.
2. Open/load that `foo.html` file in browser.
3. Open devtools and inspect the DOM shown for that `foo.html` file.

What will be seen is what’s shown in the following image:

<img width="366" height="151" alt="The DOM for a p element closed by a following ul element." src="https://github.com/user-attachments/assets/b1788201-4a72-4697-a32a-262ee62386d1" />
